src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -773,25 +778,26 @@ private void CheckAgentRootDirectorySecure()
         }
 
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA2000:Dispose objects before losing scope", MessageId = "locationServer")]
-        private async Task<bool> IsHostedServer(string serverUrl, VssCredentials credentials)
+        public static async Task<Location.ConnectionData> GetConnectionData(string serverUrl, VssCredentials credentials, ILocationServer locationServer)
         {
-            // Determine the service deployment type based on connection data. (Hosted/OnPremises)
-            var locationServer = HostContext.GetService<ILocationServer>();
             VssConnection connection = VssUtil.CreateConnection(new Uri(serverUrl), credentials);
             await locationServer.ConnectAsync(connection);
+            return await locationServer.GetConnectionDataAsync();
+        }
+
+        public static bool IsHostedServer(Location.ConnectionData connectionData)
+        {
+            // Determine the service deployment type based on connection data. (Hosted/OnPremises)
             try
             {
-                var connectionData = await locationServer.GetConnectionDataAsync();
-                Trace.Info($"Server deployment type: {connectionData.DeploymentType}");
                 return connectionData.DeploymentType.HasFlag(DeploymentFlags.Hosted);
             }
-            catch (Exception ex)
+            catch (Exception)
             {
                 // Since the DeploymentType is Enum, deserialization exception means there is a new Enum member been added.
                 // It's more likely to be Hosted since OnPremises is always behind and customer can update their agent if are on-prem
-                Trace.Error(ex);
```

Comment:
Why should we skip the error tracing?

Reply:
Made `switch...case` check for deployment type